refactor(napi/oxlint): call `lintFile` with args not array

overlookmotel opened the pull request:
Use `FnArgs` in `JsLintFileCb` to make `lintFile` JS function get called with separate arguments, rather than an array. `ThreadsafeFunction`'s API is weird!

camc314 commented:
### Merge activity

* **Jul 28, 11:15 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

overlookmotel committed:
refactor(napi/oxlint): call `lintFile` with args not array (#12568)

Use `FnArgs` in `JsLintFileCb` to make `lintFile` JS function get called with separate arguments, rather than an array. `ThreadsafeFunction`'s API is weird!
